src/axom/core/NumericLimits.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *
+ * \file NumericLimits.hpp
+ *
+ * \brief Header file containing portability layer for std::numeric_limits
+ *        capabilities
+ *
+ */
+
+#ifndef AXOM_NUMERICLIMITS_HPP_
+#define AXOM_NUMERICLIMITS_HPP_
+
+#include "axom/config.hpp"  // for compile-time definitions
+
+#include <limits>
+
+#if defined(AXOM_USE_CUDA)
+  #include <cuda/std/limits>
+#endif
+
+namespace axom
+{
+#if defined(AXOM_USE_CUDA)
+// Note: cuda::std types work in host and device code as long as Axom is
+//       configured with CUDA enabled. No need to rely on two different
+//       header files in that case.
+template <typename T>
+using numeric_limits = cuda::std::numeric_limits<T>;
```

Comment:
Is the point of using CUDA-specific numeric limits to get __host__ __device__ on them? Or, does that not matter since they are probably all constexpr? Does HIP have a header like this? Thanks.

Reply:
The main point of this is to prevent compiler warnings (calling host function from host-device function) that a few users have reported. It is not needed for HIP since the amdclang compiler is a unified host-device compiler and know how to sort everything out.